auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BasicAuthToBearerTokenFilter.java

```diff
@@ -0,0 +1,135 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tokens.auth.http;
+
+import com.google.common.io.BaseEncoding;
+import com.google.common.net.HttpHeaders;
+import com.palantir.tokens.auth.AuthHeader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link Filter} that replaces basic auth with a bearer token. It assumes the bearer token is in the password field
+ * of the basic auth header.
+ */
+public class BasicAuthToBearerTokenFilter implements Filter {
+    private static final String BASIC_AUTH_STR = "Basic";
+    private static final Logger log = LoggerFactory.getLogger(BasicAuthToBearerTokenFilter.class);
+    private static final BaseEncoding BASE_64_ENCODING = BaseEncoding.base64Url();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public final void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        ServletRequest updatedRequest;
+        try {
+            updatedRequest = getRequestWithToken(request);
+        } catch (NullRawAuthHeaderException e) {
+            updatedRequest = request;
+        } catch (AuthHeaderNotBasicAuthException e) {
+            log.warn("AuthHeaderNotBasicAuthException", e);
+            updatedRequest = request;
+        }
+        chain.doFilter(updatedRequest, response);
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    private ServletRequest getRequestWithToken(ServletRequest request)
+            throws AuthHeaderNotBasicAuthException, NullRawAuthHeaderException {
+        if (request instanceof HttpServletRequest) {
+            final HttpServletRequest httpRequest = (HttpServletRequest) request;
+            final String rawAuthHeader = httpRequest.getHeader(HttpHeaders.AUTHORIZATION);
+            return getRequestWithTokenFromRawAuthHeader(rawAuthHeader, httpRequest);
+        } else {
+            throw new AuthHeaderNotBasicAuthException("Request is not an HttpServletRequest.");
+        }
+    }
+
+    private ServletRequest getRequestWithTokenFromRawAuthHeader(String rawAuthHeader, HttpServletRequest request)
+            throws AuthHeaderNotBasicAuthException, NullRawAuthHeaderException {
+        if (rawAuthHeader == null) {
+            throw new NullRawAuthHeaderException("Raw auth header is null.");
```

Comment:
Yes would rather not have custom exceptions. The trouble is we want to respond differently to different failures: if the `rawAuthHeader` is null, we don't want to log anything. If it's not null, and it's not in basic auth format, we want to log that.

Commit `b5d03a` removes the custom exceptions. The penalty is we have some code duplication with logging an error message and then throwing an `IllegalArgumentException` with that message.